fury/ui/helpers.py

```diff
@@ -23,6 +22,81 @@ def clip_overflow(textblock, width, side='right'):
     -------
     clipped text : str
         Clipped version of the text.
+    """
+    original_str = textblock.message
+    prev_bg = textblock.have_bg
+
+    _, mid_ptr = check_overflow(textblock, width, '...', side)
+
+    if mid_ptr == 0:
+        return original_str
+
+    textblock.have_bg = prev_bg
+    return textblock.message
+
+
+def wrap_overflow(textblock, wrap_width, side='right'):
+    """Wraps overflowing text of TextBlock2D with respect to width.
+
+    Parameters
+    ----------
+    textblock : TextBlock2D
+        The textblock object whose text needs to be wrapped.
+    wrap_width : int
+        Required width of the wrapped text.
+    side : str, optional
+        Clips the overflowing text according to side.
+        It takes values "left" or "right".
+
+    Returns
+    -------
+    wrapped text : str
+        Wrapped version of the text.
+    """
+    original_str = textblock.message
+    str_copy = textblock.message
+    prev_bg = textblock.have_bg
+    wrap_idx = []
+
+    _, mid_ptr = check_overflow(textblock, wrap_width, '', side)
+
```

Comment:
I would recommend to change the name of `mid_ptr`. Maybe `wrap_idx`? and `wrap_idx` to `wrap_idxs` or something similar which is plural.

Reply:
This has been updated to avoid confusion.